src/helpers/github.rs

```diff
@@ -568,6 +568,56 @@ impl MetadataSource for GitHubReleaseAsset {
     }
 }
 
+#[allow(dead_code)]
+#[derive(PartialEq, Debug, Clone)]
```

Comment:
We might be able to sidestep the need to implement the serialization/deserialization entirely by relying on Serde's built in logic.

```suggestion
#[derive(PartialEq, Debug, Clone, Serialize, Deserialize)]
```

Then for each of the variants, we can use [`#[serde(rename="github/repo")]`](https://serde.rs/variant-attrs.html#rename) (or equivalent) to change how it is represented.

I haven't checked it yet locally, but I suspect it should work.

Reply:
good point